clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -624,19 +596,160 @@ struct Table {
   }
 };
 
+static Expected<std::vector<module_split::SplitModule>>
+parseSplitModulesFromFile(StringRef File) {
+  Table T;
+  if (Error E = T.populateSYCLTable(File))
+    return std::move(E);
+
+  std::vector<module_split::SplitModule> Modules;
+  for (auto &Entry : T.Entries) {
+    llvm::util::PropertySetRegistry Properties;
+    std::string Symbols;
+    if (!Entry.PropFile.empty()) {
+      auto MBOrErr = MemoryBuffer::getFile(Entry.PropFile);
+      if (!MBOrErr)
+        return createFileError(Entry.PropFile, MBOrErr.getError());
+
+      auto &MB = **MBOrErr;
+      auto PropSetOrErr = llvm::util::PropertySetRegistry::read(&MB);
+      if (!PropSetOrErr)
+        return PropSetOrErr.takeError();
+
+      Properties = std::move(**PropSetOrErr);
+    }
+
+    if (!Entry.SymFile.empty()) {
+      auto MBOrErr = MemoryBuffer::getFile(Entry.SymFile);
+      if (!MBOrErr)
+        return createFileError(Entry.SymFile, MBOrErr.getError());
+
+      auto &MB = *MBOrErr;
+      Symbols = std::string(MB->getBufferStart(), MB->getBufferEnd());
+    }
+
+    Modules.emplace_back(Entry.IRFile, std::move(Properties),
+                         std::move(Symbols));
+  }
+
+  return Modules;
+}
+
+// Run sycl-post-link tool
+static Expected<std::vector<module_split::SplitModule>>
```

Comment:
I suppose we will remove this path in subsequent PRs?

Reply:
Yes.